docs/source/authentication.rst

```diff
@@ -4,17 +4,36 @@
 Authentication
 **************
```

Comment:
The big thing that jumped out at me from this file in general was that it's actually logically structured, but not in a way that de-emphasizes the API keys. I.e. the text itself de-emphasizes API keys, but the structure makes them equal with user identity. 

While we're on track to move towards 'service accounts' (that is user accounts with specifically scoped permissions, rather than API keys per project/dataset etc) I think we want to move towards 'hiding' the API Keys. Towards that end I would reformat our with structure more similar to below:
```
 - Types of authentication
   - user authentication via public key (*recommended*)
   - resource authentication via api key (supports legacy flows)
 - User Authentication
   - register your key
   - use your key
 - Resource Authentication
   - get a key
   - use the key
```

And then we can slough off resource keys as we move forward

Reply:
Great point